controllers/preflightvalidationocp_reconciler.go

```diff
@@ -0,0 +1,271 @@
+	kmmv1beta1 "github.com/rh-ecosystem-edge/kernel-module-management/api/v1beta1"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/auth"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/filter"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/registry"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/statusupdater"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/syncronizedmap"
+)
+
+const (
+	driverToolkitSpecName        = "driver-toolkit"
+	driverToolkitJSONFilePath    = "etc/driver-toolkit-release.json"
+	releaseManifestImagesRefFile = "release-manifests/image-references"
+)
+
+type dtkRelease struct {
+	KernelVersion   string `json:"KERNEL_VERSION"`
+	RTKernelVersion string `json:"RT_KERNEL_VERSION"`
+	RHELVersion     string `json:"RHEL_VERSION"`
+}
+
+// PreflightValidationOCPReconciler reconciles a PreflightValidationOCP object
+type PreflightValidationOCPReconciler struct {
+	client             client.Client
+	filter             *filter.Filter
+	registry           registry.Registry
+	registryAuthGetter auth.RegistryAuthGetter
+	kernelOsDtkMapping syncronizedmap.KernelOsDtkMapping
+	statusUpdater      statusupdater.PreflightOCPStatusUpdater
+	scheme             *runtime.Scheme
+}
+
+func NewPreflightValidationOCPReconciler(
+	client client.Client,
+	filter *filter.Filter,
+	registry registry.Registry,
+	authFactory auth.RegistryAuthGetterFactory,
+	kernelOsDtkMapping syncronizedmap.KernelOsDtkMapping,
+	statusUpdater statusupdater.PreflightOCPStatusUpdater,
+	scheme *runtime.Scheme) *PreflightValidationOCPReconciler {
+	registryAuthGetter := authFactory.NewClusterAuthGetter()
+	return &PreflightValidationOCPReconciler{
+		client:             client,
+		filter:             filter,
+		registry:           registry,
+		registryAuthGetter: registryAuthGetter,
+		kernelOsDtkMapping: kernelOsDtkMapping,
+		statusUpdater:      statusUpdater,
+		scheme:             scheme,
+	}
+}
+
+func (r *PreflightValidationOCPReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("preflightvalidationocp").
+		For(&kmmv1beta1.PreflightValidationOCP{}, builder.WithPredicates(filter.PreflightOCPReconcilerUpdatePredicate())).
+		Watches(
+			&source.Kind{Type: &kmmv1beta1.PreflightValidation{}},
+			handler.EnqueueRequestsFromMapFunc(r.filter.EnqueueAllPreflightValidationsOCP),
+		).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: 1,
+		}).
+		Complete(r)
+}
+
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidations,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidations/status,verbs=get;list;watch
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidations/finalizers,verbs=update
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidationsocp,verbs=get;list;watch;create;update;patch;delete
```

Comment:
If I'm not mistaken we could remove `create` and `delete` permissions on `PreflightValidationsOCP` CRs, as only the users would perform such operations.  WDYT?

Reply:
fixed